### DISCUSSION TITLE
Finding and showing borders between watershed regions

### DISCUSSION THREAD

**Discussion by artizzq** [PMEAL/porespy#711] [Q&A]
Greetings. Could you provide python code snippet how to create borders between segments. I havent found a way to do it, please help. 

![image](https://user-images.githubusercontent.com/56273839/183566617-bd77940b-717a-48cf-a7aa-dda1a2c68ba3.png)
Versatile and efficient pore network extraction method using marker-based watershed segmentation - Scientific Figure on ResearchGate. Available from: https://www.researchgate.net/figure/Illustration-of-marker-based-watershed-segmentation-procedure-on-a-simple-2D-image-of_fig2_319167819 [accessed 9 Aug, 2022]

And how do I create points on the middle of those borders (added like this below):
![image](https://user-images.githubusercontent.com/56273839/183567402-0ead3a57-8097-408b-b417-29715a139222.png)

**Comment by jgostick** [comment]
For the borders, there are 2 options.  You can tell the watershed to return borders, but this creates a different segmentation for some reason.  They I way did it was to scan through each region, make it a boolean, then dilate it.  Then keep only the dilation (subtract the original region).  Finally, you can add each dilated region to a master image or mask, then invert it, and multiply the dt by this mask...voila, you'll have the dark purple lines.

For the points, you can again do it multiple way.  Try getting the skeleton and multiplying the mask obtained above by the skeleton.

**Reply by artizzq** [reply]
I tried myself but no avail https://github.com/PMEAL/porespy/discussions/711#discussioncomment-3359446

**Reply by artizzq** [reply]
Figured out. The question is closed.

**Comment by artizzq** [comment]
@jgostick, I made this code
```
    sample = np.zeros_like(snow_out.regions, dtype='f')
    for i, region in enumerate(snow_out.regions):
        snow_out.regions[i] = snow_out.regions[i].astype(bool)
        result = spim.binary_dilation(im, structure=disk(3)) - snow_out.regions[i]
        sample += np.invert(result)

    sample *= snow_out.dt
```
but I got this result which isnt what I expect to see

![image](https://user-images.githubusercontent.com/56273839/183701355-0b621219-6735-4603-8225-9fc15beb925f.png)

And I tried
```
    sample = snow_out.regions
    for i, region in enumerate(snow_out.regions):
        sample += ps.filters.fftmorphology(snow_out.regions[i].astype(bool), strel=disk(4), mode='dilation') - snow_out.regions[i]
    sample = np.invert(sample)

    sample *= snow_out.dt
```
This happens
```
  File "...test.py", line 1322, in <module>
    sample += ps.filters.fftmorphology(snow_out.regions[i].astype(bool), strel=disk(4), mode='dilation') - snow_out.regions[i]
  File "...lib\site-packages\porespy\filters\_fftmorphology.py", line 82, in fftmorphology
    temp = dilate(temp, strel)
  File "...lib\site-packages\porespy\filters\_fftmorphology.py", line 71, in dilate
    t = fftconvolve(im, strel, mode='same') > 0.1
  File "...lib\site-packages\scipy\signal\_signaltools.py", line 654, in fftconvolve
    raise ValueError("in1 and in2 should have the same dimensionality")
ValueError: in1 and in2 should have the same dimensionality
```

I dont know what I am doing wrong

UPD:
I wrote another code that suppose to do something I want
```
    sample = np.zeros_like(snow_out.regions)
    sample1 = snow_out.dt
    for i, region in enumerate(snow_out.regions):
        sample += np.invert(spim.binary_dilation(region.astype(bool)) - region)
    sample1 *= sample

```

But again I dont see any lines or something near it
![image](https://user-images.githubusercontent.com/56273839/183713745-7927ec88-f0b9-45e6-a378-5fa04aed21f2.png)

UPD:

Figured out myself. Closed.

**Reply by jgostick** [reply]
I knew you would!  Would mind pasting your code here for future users?
